File: packages/android_intent_plus/lib/src/put_classes/put_long.dart
import 'package:android_intent_plus/src/put_classes/base/put_base.dart';

class PutLong extends PutBase<int> {
  PutLong({required String key, required int value})
      : super(key: key, value: value);

  @override
  String get javaClass => 'PutLong';

  factory PutLong.fromJson({required String key, required dynamic value}) {
    return PutLong(key: key, value: value as int);
  }

  @override
  Map<String, dynamic> toJson() {
    return <String, dynamic>{
      'key': key,
      'javaClass': javaClass,
      'value': value,
    };
  }
}

File: packages/android_intent_plus/android/src/main/java/dev/fluttercommunity/plus/androidintent/Bundle/PutClasses/PutLong.java
package dev.fluttercommunity.plus.androidintent.Bundle.PutClasses;

import android.os.Bundle;
import dev.fluttercommunity.plus.androidintent.Bundle.Constants;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.base.PutBase;
import org.json.JSONException;
import org.json.JSONObject;

public class PutLong extends PutBase<Long> {

  public static final String JAVA_CLASS = "PutLong";

  public PutLong(String key, Long value) {
    super(key, JAVA_CLASS, value);
  }

  public static void convert(Bundle bundle, PutLong putLong) {
    bundle.putLong(putLong.key, putLong.value);
  }

  public static PutLong fromJson(JSONObject jsonObject) throws JSONException {
    return new PutLong(jsonObject.getString(Constants.KEY), jsonObject.getLong(Constants.VALUE));
  }

  @Override
  public JSONObject toJson() throws JSONException {
    JSONObject jsonObject = new JSONObject();
    jsonObject.put("key", key);
    jsonObject.put("javaClass", JAVA_CLASS);
    jsonObject.put("value", value);
    return jsonObject;
  }
}
